### DISCUSSION TITLE
Future

### DISCUSSION THREAD

**Discussion by MgSam** [microsoft/DacFx#95] [General] [answered]
I was glad to read in the [recent blog post](https://techcommunity.microsoft.com/t5/azure-sql-blog/microsoft-build-sql-the-next-frontier-of-sql-projects/ba-p/3290628) that the old .sqlproj format is finally being replaced. Looking forward to this working in VS / SSDT as well.

I have several questions about this project and related technologies:
- Does this repo cover Dacpacs as well as the SQL project format? Like others, I'd love to see dacpac technology be open sourced and modernized. For example, Dacpacs still do not support CREATE OR ALTER syntax over 6 years after SQL Server added support for such. 
- If at all, how does this relate to SMO? I believe dacpac has been implemented historically using SMO. Given all the many problems with SMO, I'd love to see it replaced with a modern, async, (unit tested!), thoughtfully designed API. I may be interested in contributing to such a project if it was open source.

**Comment by dzsquared** [comment, marked as answer]
Good questions!
This repo does cover discussions around all of the data-tier application framework (dacfx), including building dacpacs.  We have not open-sourced the full dacfx library at this time but are continuing to invest in it with new features and work on a road to being open source.

For SMO - SQL Management Objects is open source and is in its own repository: https://github.com/microsoft/sqlmanagementobjects
You're welcome to contribute to the unit tests (https://github.com/microsoft/sqlmanagementobjects/tree/main/src/UnitTest) or other areas of the library.
(DacFx is not implemented with SMO)
